jerry-ext/include/jerryscript-ext/section.h

```diff
@@ -0,0 +1,81 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef JERRY_EXT_SECTION_H
```

Comment:
I've seen that discussion.

> The problem is that context.h is public whereas section-utils.h is private.
> So, I guess I should publish section-utils.h - maybe as jerryscript-ext/section.h - because being able to easily manipulate sections seems like it might be beneficial.

Beneficial, but only for the purpose of implementing extensions, not for users of extensions.

> Who says you can't have a header-only extension, eh? :)
 
Consider this as kind of a doubt :)